File: src/java/magma/Placeholder.java
package magma;

public record Placeholder(String value) implements Definable {
    static String wrap(final String input) {
        return "/*" + input.replace("/*", "start").replace("*/", "end") + "*/";
    }

    @Override
    public String generate() {
        return Placeholder.wrap(this.value);
    }
}

File: src/node/magma/Placeholder.ts

class Placeholder {
	constructor () {
	}
	/*private final String value;*/
	constructor (value : string) {
		this.value = value;
	}
	generate(input : string) : string {
		let replaced : any = input.replace("/*", "start").replace("*/", "end");
		return "/*" + replaced + "*/";
	}
	generateWithAfterName(afterName : string) : string {
		return Placeholder.generate(this.value) + afterName;
	}
	generate() : string {
		return this.value;
	}
}

